iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -1089,153 +1090,634 @@ TYPED_TEST(stringTyped_test, CompareOperatorGreaterEqResultFalseWithDifferentCap
+    string<STRINGCAP + 3U> sut;
+    ASSERT_THAT(sut.unsafe_assign(temp), Eq(true));
+
+    char testChar1 = 'L';
+    EXPECT_THAT(sut.compare(testChar1), Gt(0));
+    char testChar2 = 'M';
+    EXPECT_THAT(sut.compare(testChar2), Gt(0));
+}
+
+/// @note bool operator==(const char& rhs) const noexcept
+/// bool operator!=(const char& rhs) const noexcept
+TYPED_TEST(stringTyped_test, CheckForEqualityWithEqualCharWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "d4e79679-939f-4aa9-9db1-4b3ba38b5dc7");
+    this->testSubject = "M";
+
+    const char testChar = 'M';
+    EXPECT_THAT(this->testSubject == testChar, Eq(true));
+    EXPECT_THAT(testChar == this->testSubject, Eq(true));
+    EXPECT_THAT(this->testSubject != testChar, Eq(false));
+    EXPECT_THAT(testChar != this->testSubject, Eq(false));
+}
+
+TYPED_TEST(stringTyped_test, CheckForEqualityWithUnequalCharWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "1547d1dd-bf34-45ec-be4e-64fdc7164404");
+    this->testSubject = "M";
+
+    char testChar = 'L';
+    EXPECT_THAT(this->testSubject == testChar, Eq(false));
+    EXPECT_THAT(testChar == this->testSubject, Eq(false));
+    EXPECT_THAT(this->testSubject != testChar, Eq(true));
+    EXPECT_THAT(testChar != this->testSubject, Eq(true));
+}
+
+TYPED_TEST(stringTyped_test, CheckForEqualityWithCharWithDifferentSizeWorks)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "d3ceaf9a-6f77-4688-be70-9a56a7b2cb78");
+    ::testing::Test::RecordProperty("TEST_ID", "c49cc0a3-b31c-49ec-90e2-707216df5eaa");
     using MyString = typename TestFixture::stringType;
     constexpr auto STRINGCAP = MyString::capacity();
-    string<STRINGCAP> testFixedString("M");
-    std::string testSTDString = "Müslimädchen";
-    EXPECT_THAT(testFixedString != testSTDString, Eq(true));
+    std::string temp(STRINGCAP + 4U, 'M');
+    string<STRINGCAP + 4U> sut;
+    ASSERT_THAT(sut.unsafe_assign(temp), Eq(true));
+
+    char testChar = 'M';
+    EXPECT_THAT(sut == testChar, Eq(false));
+    EXPECT_THAT(testChar == sut, Eq(false));
+    EXPECT_THAT(sut != testChar, Eq(true));
+    EXPECT_THAT(testChar != sut, Eq(true));
 }
 
-TYPED_TEST(stringTyped_test, CompareOperatorFixedStringNotEqualSTDStringWithSameSizeResultTrue)
+/// @note bool operator<(const char& rhs) const noexcept
+/// bool operator<=(const char& rhs) const noexcept
+/// bool operator>(const char& rhs) const noexcept
+/// bool operator>=(const char& rhs) const noexcept
+TYPED_TEST(stringTyped_test, CompareOperatorsWithDifferentChar)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "99a40704-9d0b-49f9-bd5c-4c87b9f2ba00");
+    ::testing::Test::RecordProperty("TEST_ID", "c818b150-b926-4f3c-8405-6327303f12f6");
     using MyString = typename TestFixture::stringType;
     constexpr auto STRINGCAP = MyString::capacity();
-    string<STRINGCAP> testFixedString("M");
-    std::string testSTDString = "L";
-    EXPECT_THAT(testFixedString != testSTDString, Eq(true));
+    string<STRINGCAP> sut1("M");
```

Comment:
could you rename this into `sutGreaterTestChar` and `sutSmallerTestChar` so that the `EXPECT_` condition can be easier reviewed.

Reply:
Please do this also for the tests above where it is applicable.